### PR TITLE
Docs/e2e sample

### PR DESCRIPTION
**Pull request by norelina**
Added docs for Chapter 6: Deploy and run e2e sample

### CODE REVIEW

File: docs/onboarding/06-e2e-sample.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+The following exercises will help you build an end-to-end solution using the [Tutorial: Build out an end-to-end solution](https://docs.microsoft.com/en-us/azure/digital-twins/tutorial-end-to-end) as a guide.
+
+1. Configure the sample project for your Azure Digital Twin instance
+2. Learn about the sample building scenario and run the sample client app
+3. Process simulated telemetry from an IoT Hub device using an Azure Function app
+4. Propagate changes through the twin graph, by processing digital twin notifications with Azure Functions, endpoints, and routes
+
+## Things to Consider
+
```
Comment (TechnicallyWilliams) [inline docs/onboarding/06-e2e-sample.md:25]: Suggestion:
 - Is there an event signaling service that might act as an alternative to Azure Event Grid when working with Azure Digital Twins? If so, how would integration work?

File: docs/onboarding/06-e2e-sample.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+## Prerequisites
+
+In addition to the Prerequisites defined in the [README](README.md), the following steps are required to complete the learning exercises below:
+
+- [Set up your session](https://docs.microsoft.com/en-us/azure/digital-twins/tutorial-end-to-end#set-up-cloud-shell-session) and register your Azure Subscription with the Azure Digital Twins namespace
+- [Add the Azure IoT Extension](https://docs.microsoft.com/en-us/azure/digital-twins/how-to-use-cli#get-the-extension) for Azure CLI
+- [Create an Azure Digital Twin](https://docs.microsoft.com/en-us/azure/digital-twins/how-to-set-up-instance-cli#create-the-azure-digital-twins-instance) instance
+- Download the sample project [digital-twins-samples](https://docs.microsoft.com/en-us/azure/digital-twins/tutorial-end-to-end#get-required-resources)
+
```
Comment (TechnicallyWilliams) [inline docs/onboarding/06-e2e-sample.md:14]: > Note: Use the Visual Studio Installer to ensure your installation of Visual Studio 2019 includes the _ASP.NET and web development_ workload/framework. Without this, an Azure Function cannot be published.

File: docs/onboarding/06-e2e-sample.md (hunk 1 above)
Comment (TechnicallyWilliams) [inline docs/onboarding/06-e2e-sample.md:25]: Suggestion:
- How might work be divided between a group of developers building an e2e Azure Digital Twins solution?

Comment (TechnicallyWilliams) [inline docs/onboarding/06-e2e-sample.md:25]: Suggestion:
- What are some other events not included in the e2e sample that an Azure Digital Twins instance might subscribe to?

Comment (norelina) [reply docs/onboarding/06-e2e-sample.md:25]: Thank you Dexter! Added.

Comment (norelina) [reply docs/onboarding/06-e2e-sample.md:25]: Thank you Dexter! Added.

Comment (norelina) [reply docs/onboarding/06-e2e-sample.md:25]: Added.

File: docs/onboarding/06-e2e-sample.md (hunk 2 above)
Comment (norelina) [reply docs/onboarding/06-e2e-sample.md:14]: Good catch. Thank you Dexter! Included note.

Comment (TechnicallyWilliams) [reply docs/onboarding/06-e2e-sample.md:14]: Correction: 
> [!NOTE] 
> Ensure your installation of Visual Studio 2019 includes the **[Azure Development Workload](https://docs.microsoft.com/en-us/dotnet/azure/configure-visual-studio)**. This workload enables the application to publish the SampleFunctionApp.

Comment (norelina) [reply docs/onboarding/06-e2e-sample.md:14]: Thank you for double checking and testing this!! :)

File: docs/onboarding/06-e2e-sample.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Chapter 6: Deploy and run end-to-end sample
+
+The following docs will introduce an end-to-end solution where an Azure Digital Twins instance is connected to other Azure services for management of devices and data.
```
Comment (charleszipp) [inline docs/onboarding/06-e2e-sample.md:3]: Aside from the tutorial, is there any other content you have encountered that describes how e2e solutions are typically architected for ADT? For example, are there any public customer use case videos that describe their architecture? Is there any content beyond the docs.microsoft.com that might provides some different perspectives?

This part of the doc should include links to any reading or videos that teach the concepts that will need to be applied in the challenge.

File: docs/onboarding/06-e2e-sample.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Key Points
+
+- Create an end-to-end scenario that shows Azure Digital Twins being driven by live device data.
```
Comment (charleszipp) [inline docs/onboarding/06-e2e-sample.md]: Rather than a description of what the reader will be doing as part of this challenge, use this section to summarize key lessons that should be understood before moving forward into the challenge. For example...
- Digital twins are created and updated in an ingestion pipeline that sources data from real-world devices.
- Azure Functions can facilitate the transformation of device data to digital twin properties and telemetry

File: docs/onboarding/06-e2e-sample.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+- Understand how changes are propagated through the twin graph by processing digital twin notifications.
+
+## Prerequisites
+
+In addition to the Prerequisites defined in the [README](README.md), the following steps are required to complete the learning exercises below:
+
+- [Set up your session](https://docs.microsoft.com/en-us/azure/digital-twins/tutorial-end-to-end#set-up-cloud-shell-session) and register your Azure Subscription with the Azure Digital Twins namespace
```
Comment (charleszipp) [inline docs/onboarding/06-e2e-sample.md]: Should we move this to the root onboarding readme? This will probably be needed for other docs as well

File: docs/onboarding/06-e2e-sample.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+- [Add the Azure IoT Extension](https://docs.microsoft.com/en-us/azure/digital-twins/how-to-use-cli#get-the-extension) for Azure CLI
```
Comment (charleszipp) [inline docs/onboarding/06-e2e-sample.md]: Add a mention that this step should have been completed in (i believe) the first learning exercise. Maybe add link back to that too.

File: docs/onboarding/06-e2e-sample.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+- [Create an Azure Digital Twin](https://docs.microsoft.com/en-us/azure/digital-twins/how-to-set-up-instance-cli#create-the-azure-digital-twins-instance) instance
```
Comment (charleszipp) [inline docs/onboarding/06-e2e-sample.md]: They should be doing this as part of the learning exercise and tutorial right? If so do we need to have it as a prereq?

File: docs/onboarding/06-e2e-sample.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+- Download the sample project [digital-twins-samples](https://docs.microsoft.com/en-us/azure/digital-twins/tutorial-end-to-end#get-required-resources)
+- Ensure your installation of Visual Studio 2019 includes the [Azure Development workload](https://docs.microsoft.com/en-us/dotnet/azure/configure-visual-studio). This workload enables the application to publish Azure Function Apps.
```
Comment (charleszipp) [inline docs/onboarding/06-e2e-sample.md]: Should this be part of the pre-reqs in the root onboarding readme?

File: docs/onboarding/06-e2e-sample.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+- Understand how changes are propagated through the twin graph by processing digital twin notifications.
+
+## Prerequisites
+
+In addition to the Prerequisites defined in the [README](README.md), the following steps are required to complete the learning exercises below:
+
+- [Set up your session](https://docs.microsoft.com/en-us/azure/digital-twins/tutorial-end-to-end#set-up-cloud-shell-session) and register your Azure Subscription with the Azure Digital Twins namespace
+- [Add the Azure IoT Extension](https://docs.microsoft.com/en-us/azure/digital-twins/how-to-use-cli#get-the-extension) for Azure CLI
+- [Create an Azure Digital Twin](https://docs.microsoft.com/en-us/azure/digital-twins/how-to-set-up-instance-cli#create-the-azure-digital-twins-instance) instance
+- Download the sample project [digital-twins-samples](https://docs.microsoft.com/en-us/azure/digital-twins/tutorial-end-to-end#get-required-resources)
```
Comment (charleszipp) [inline docs/onboarding/06-e2e-sample.md]: Same as above, the reader will need to do this as part of the learning exercise. If so do we need list it as a prereq?

File: docs/onboarding/06-e2e-sample.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+- Ensure your installation of Visual Studio 2019 includes the [Azure Development workload](https://docs.microsoft.com/en-us/dotnet/azure/configure-visual-studio). This workload enables the application to publish Azure Function Apps.
+
+## Learning Exercises
+
+The following exercises will help you build an end-to-end solution using the [Tutorial: Build out an end-to-end solution](https://docs.microsoft.com/en-us/azure/digital-twins/tutorial-end-to-end) as a guide.
+
+1. Configure the sample project for your Azure Digital Twin instance
```
Comment (charleszipp) [inline docs/onboarding/06-e2e-sample.md]: I would reword this list as acceptance criteria as opposed to list of tasks. What do you expect the reader to be able to demonstrate at the end of the exercise? What would their demo look like? This will give the reader more agency in how they solve it and potentially yield novel approaches and feedback that we didn't anticipate and may not be documented well.

File: docs/onboarding/06-e2e-sample.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+2. Learn about the sample building scenario and run the sample client app
+3. Process simulated telemetry from an IoT Hub device using an Azure Function app
+4. Propagate changes through the twin graph, by processing digital twin notifications with Azure Functions, endpoints, and routes
+
+## Things to Consider
```
Comment (charleszipp) [inline docs/onboarding/06-e2e-sample.md:24]: How about one for testing? How would you prove that the solution implements desired capabilities?

Comment (charleszipp) [inline docs/onboarding/06-e2e-sample.md:24]: Use this as an opportunity to have the reader think about how they would apply our engineering fundamentals. For example

> -What steps would you take, if any, to strengthen the security of the solution? 

The troubleshooting and deploymnt points definitely do this BTW.

**Review by charleszipp — CHANGES_REQUESTED** [review]

File: docs/onboarding/06-e2e-sample.md (hunk 8 above)
Comment (TechnicallyWilliams) [reply docs/onboarding/06-e2e-sample.md]: This pre-reqs notes has been merged into Microsoft docs master [here](https://github.com/MicrosoftDocs/azure-docs-pr/blob/master/includes/digital-twins-tutorial-sample-prereqs.md)

We can now exclude it from our walkthrough.

Comment (TechnicallyWilliams) [reply docs/onboarding/06-e2e-sample.md]: Looks like some git ops going on. I just realized my merge went to MicrosoftDocs/azure-docs-pr master, **NOT** MicrosoftDocs/azure-docs master. 

It'll take some time before it shows up in the public version of the docs [here](https://docs.microsoft.com/en-us/azure/digital-twins/tutorial-end-to-end#prerequisites)

File: docs/onboarding/06-e2e-sample.md (hunk 11 above)
Comment (norelina) [reply docs/onboarding/06-e2e-sample.md:24]: Yes! Thank you @charleszipp . Added

Comment (norelina) [reply docs/onboarding/06-e2e-sample.md:24]: Adjusted section. Thank you for the insight. I will leave this comment  unresolved until re-review.

File: docs/onboarding/06-e2e-sample.md (hunk 10 above)
Comment (norelina) [reply docs/onboarding/06-e2e-sample.md]: Adjusted verbiage on steps. I will leave this comment unresolved until re-review.

File: docs/onboarding/06-e2e-sample.md (hunk 9 above)
Comment (norelina) [reply docs/onboarding/06-e2e-sample.md]: Removed this section. You are right, it ended up being duplicated content.

File: docs/onboarding/06-e2e-sample.md (hunk 8 above)
Comment (norelina) [reply docs/onboarding/06-e2e-sample.md]: Removed Prerequisites section from this page and adjusted prerequisites on README. We can re-refine when doc update is published. Thank you both!

File: docs/onboarding/06-e2e-sample.md (hunk 7 above)
Comment (norelina) [reply docs/onboarding/06-e2e-sample.md]: Removed this section. You are right, it ended up being duplicated content.

File: docs/onboarding/06-e2e-sample.md (hunk 6 above)
Comment (norelina) [reply docs/onboarding/06-e2e-sample.md]: Removed this section. You are right, it ended up being duplicated content.

File: docs/onboarding/06-e2e-sample.md (hunk 5 above)
Comment (norelina) [reply docs/onboarding/06-e2e-sample.md]: Done. Thank you!

File: docs/onboarding/06-e2e-sample.md (hunk 4 above)
Comment (norelina) [reply docs/onboarding/06-e2e-sample.md]: Adjusted, but will leave comment unresolved until re-review

File: docs/onboarding/06-e2e-sample.md (hunk 3 above)
Comment (norelina) [reply docs/onboarding/06-e2e-sample.md:3]: I struggled with that one. I ended up adding some links to more complete examples for ingesting from IoT hub, and for managing endpoints and routes. Any other ideas on appropriate content would be welcome. @TechnicallyWilliams  @charleszipp

**Review by charleszipp — APPROVED** [review]
